crates/ruff_python_ast/src/parenthesize.rs

```diff
@@ -0,0 +1,42 @@
+use std::ops::Sub;
+
+use ruff_python_trivia::{SimpleTokenKind, SimpleTokenizer};
+use ruff_text_size::{TextRange, TextSize};
+
+use crate::node::AnyNodeRef;
+use crate::{ExpressionRef, Ranged};
+
+/// Returns the [`TextRange`] of a given expression including parentheses, if the expression is
+/// parenthesized; or `None`, if the expression is not parenthesized.
+pub fn parenthesized_range(
+    expr: ExpressionRef,
+    parent: AnyNodeRef,
+    contents: &str,
+) -> Option<TextRange> {
+    // If the parent is a node that brings its own parentheses, exclude the closing parenthesis
+    // from our search range. Otherwise, we risk matching on calls, like `func(x)`.
+    // We can exclude tuples here, since the only risk is a single-element tuple (e.g. `(x,)`),
+    // which must have a trailing comma anyway.
+    let exclusive_parent_end = if parent.is_arguments() || parent.is_parameters() {
```

Comment:
`parameters` shouldn't be a problem (or we have one in the formatter) because the parameter isn't an expression, it's a `Parameter` node and the default expression is guaranteed to be preceded by a name, making it unambiguous.